WordPress/gutenberg
# Button Component: Update styling to avoid relying on Core Styles making it reusable

Right now, the button component is relying heavily on core styles. This has some issues:

 - The need to add `.core-ui` to any place where we want to extend the default button styles
 - The components module is not reusable. One of the goals of the Gutenberg modules is to make them reusable in WP Admin and outside WP Admin as well. If we ever want to publish the components module on npm, it needs to define its own styles.
 - In components heavy application, we don't reuse classNames, we reuse components. 

This PR copies the styles from Core into the `components-button` className to solve this.

**Notes**

I probably need to update the default styles according to the WP theme cc @jasmussen (I might need some help here)

**Testing instructions**

Navigate in the Gutenberg UI and check that the styling of the different buttons didn't regress.
